staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -354,46 +346,15 @@ func (c *Config) SkipComplete() completedConfig {
 	return completedConfig{c}
 }
 
-// New returns a new instance of GenericAPIServer from the given config.
-// Certain config fields will be set to a default value if unset,
-// including:
-//   ServiceClusterIPRange
-//   ServiceNodePortRange
-//   MasterCount
-//   ReadWritePort
-//   PublicAddress
-// Public fields:
-//   Handler -- The returned GenericAPIServer has a field TopHandler which is an
-//   http.Handler which handles all the endpoints provided by the GenericAPIServer,
-//   including the API, the UI, and miscellaneous debugging endpoints.  All
-//   these are subject to authorization and authentication.
-//   InsecureHandler -- an http.Handler which handles all the same
-//   endpoints as Handler, but no authorization and authentication is done.
-// Public methods:
-//   HandleWithAuth -- Allows caller to add an http.Handler for an endpoint
-//   that uses the same authentication and authorization (if any is configured)
-//   as the GenericAPIServer's built-in endpoints.
-//   If the caller wants to add additional endpoints not using the GenericAPIServer's
-//   auth, then the caller should create a handler for those endpoints, which delegates the
-//   any unhandled paths to "Handler".
-func (c completedConfig) New() (*GenericAPIServer, error) {
-	s, err := c.constructServer()
-	if err != nil {
-		return nil, err
-	}
-
-	return c.buildHandlers(s, nil)
-}
-
-func (c completedConfig) constructServer() (*GenericAPIServer, error) {
+func (c completedConfig) constructServer(delegate http.Handler) (*GenericAPIServer, error) {
```

Comment:
please add a godoc how this is supposed to differ from New. As they have the same signature now, it's getting confusing.